### PR TITLE
Fix httpclient timeout on AzureMetada retrieving 

### PR DESCRIPTION
**Pull request by ismayilov-ismayil**
Prob: Agent contains http client which fetchs metadata from Azure to define where agent runs. It delays for a 100 seconds (default http client timeout) if Azure metada endpoint is not reachable.

Solution: Http default timeout is decreased to 5 sec. IO operation moved to another task to not block step preparation process.

Related issue: https://github.com/microsoft/azure-pipelines-agent/issues/4215

### CODE REVIEW

File: src/Agent.Sdk/Util/AzureInstanceMetadataProvider.cs [hunk 1]
```diff
@@ -16,6 +16,7 @@ class AzureInstanceMetadataProvider : IDisposable
         public AzureInstanceMetadataProvider()
         {
             _client = new HttpClient();
+            _client.Timeout = TimeSpan.FromSeconds(5);
```
[the remark this answers is not in the document]
Comment (KonstantinTyukalov) [reply src/Agent.Sdk/Util/AzureInstanceMetadataProvider.cs:19]: Oh, I see it's only for metadata provider calls, right?

[the remark this answers is not in the document]
Comment (ismayilov-ismayil) [reply src/Agent.Sdk/Util/AzureInstanceMetadataProvider.cs:19]: yes, it is only for metadata retriver client

File: src/Agent.Worker/StepsRunner.cs [hunk 2]
```diff
@@ -50,6 +52,21 @@ public async Task RunAsync(IExecutionContext jobContext, IList<IStep> steps)
             CancellationTokenRegistration? jobCancelRegister = null;
             int stepIndex = 0;
             jobContext.Variables.Agent_JobStatus = jobContext.Result ?? TaskResult.Succeeded;
+            // Wait till all async commands finish.
+            foreach (var command in jobContext.AsyncCommands ?? new List<IAsyncCommandContext>())
```
Comment (KonstantinTyukalov) [inline src/Agent.Worker/StepsRunner.cs:56]: [Question] Are we sure we want to wait all async commands here?
Probably, should we wait only specific commands?

Comment (ismayilov-ismayil) [reply src/Agent.Worker/StepsRunner.cs:56]: actually on my first approach I added boolean parameter to define which commands we should wait. but as there is not another commands except current one, I removed this parameter.

**Review by KonstantinTyukalov — APPROVED** [review]

**Review by vmapetr — APPROVED** [review]

**Review by kirill-ivlev — APPROVED** [review]

**Review by PavloAndriiesh — APPROVED** [review]

**Comment by kirill-ivlev** [comment]
LGTM, thanks!
Tested locally and validated that it is not blocking job execution and don't produce other issues.

**Pull request merged**